pipeline/pipeline/pipeline.py

```diff
@@ -92,6 +92,12 @@ def read_input_group(self, **kwargs):
         return rg
 
     def write_output(self, resource, dest):  # pylint: disable=R0201
+        assert isinstance(resource, Resource)
+        if isinstance(resource, TaskResourceFile) and resource not in resource._source._mentioned:
```

Comment:
TaskResourceFile is a Resource, so we don't need this check.

From a higher level, it seems we don't need either the assert above, or `if isinstance()`, because if resource is in resource._source._mentioned it should be a Resource.